src/core/DynamicPathTracingSceneGenerator.js

```diff
@@ -49,7 +50,17 @@ export class DynamicPathTracingSceneGenerator {
 
 				}
 
-			} );
+			}
+
+			if ( scene instanceof Object3D ) {
+
+				scene.traverse( checkObject );
+
+			} else if ( scene instanceof Array ) {
+
+				scene.forEach( checkObject );
+
+			}
 
```

Comment:
For the DynamicPathtracingSceneGenerator I think it's simplest if we just ensure that the `.scene` field is an array in the constructor which will simplify this "generate" function:

```js
this.scene = Array.isArray( scene ) ? scene : [ scene ];
```

We can also rename `this.scene` to `this.objects` so it's more semantically correct.